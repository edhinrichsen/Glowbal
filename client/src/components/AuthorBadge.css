.authorBadge {
    /* background-color: aqua; */
    /* float: left; */
    text-align: left;

    display: flex;           
    flex-direction: row; 
    justify-content: left; 
    align-items: center; 
}

.profilePicture{
    background-color: #40a9ff;
    width: 48px;
    height: 48px;
    border-radius: 64px;
    padding: 8px;
}

.authorName{
    margin-left: 16px;
    /* vertical-align: middle; */
}