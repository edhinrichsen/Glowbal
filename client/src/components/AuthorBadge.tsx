import React from 'react';

import './AuthorBadge.css';

import { RightOutlined, UserOutlined } from '@ant-design/icons';
import { Link } from 'react-router-dom';

type PortfolioObjectProps = {
  /** Author's name to be included */
  author: string;
  /** Tagline about the author to be included under the name */
  tagline?: string;
  /** A link to a profile picture */
  profilePictureLink?: string;
  /** userID for the author used for link */
  userID: string;
};

function AuthorBadge(props: PortfolioObjectProps) {
  return (
    <Link to={`/profile/${props.userID}`}>
      <div className="authorBadge">
        <div
          className="profilePicture"
          style={{
            backgroundImage: `${
              props.profilePictureLink
                ? `url(${props.profilePictureLink})`
                : undefined
            }`,
            backgroundPosition: 'center',
            backgroundSize: 'cover',
          }}
        >
          {!props.profilePictureLink ? (
            <UserOutlined style={{ fontSize: '32px', color: '#fff' }} />
          ) : null}
        </div>
        <div className="authorName">
          <h4>
            <strong>{props.author}</strong>
          </h4>
          <h5>{props.tagline}</h5>
        </div>
        <div className="gotoProfile">
          <RightOutlined style={{ color: 'grey' }} />
        </div>
      </div>
    </Link>
  );
}

export default AuthorBadge;
