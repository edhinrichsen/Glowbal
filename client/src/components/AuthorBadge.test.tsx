import React from 'react';
import { shallow } from 'enzyme';
import AuthorBadge from './AuthorBadge';

describe('Testing AuthorBadge', () => {
  it('renders without crashing', () => {
    const component = shallow(
      <AuthorBadge author="author" tagline="tagline" />
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
