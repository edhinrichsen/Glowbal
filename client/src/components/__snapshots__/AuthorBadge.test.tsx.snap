// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AuthorBadge renders without crashing 1`] = `
<div
  className="authorBadge"
>
  <div
    className="profilePicture"
    style={
      Object {
        "backgroundImage": "undefined",
        "backgroundPosition": "center",
        "backgroundSize": "cover",
      }
    }
  >
    <ForwardRef(UserOutlined)
      style={
        Object {
          "color": "#fff",
          "fontSize": "32px",
        }
      }
    />
  </div>
  <div
    className="authorName"
  >
    <h4>
      <strong>
        author
      </strong>
    </h4>
    <h5>
      tagline
    </h5>
  </div>
</div>
`;
